src/components/spinner/Spinner.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { keyframes } from '@emotion/react';
+import styled from '@emotion/styled';
+import { ReactNode } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { screenReaderOnly } from '~common/helpers/styles';
+
+interface Props {
+  className?: string;
+  children?: ReactNode;
+  label?: ReactNode;
```

Comment:
NIT: label is a react node but we are using it as a boolean. Also, I feel label prop for a spinner might be confusing?